src/server/svr_func.c

```diff
@@ -5277,6 +5305,41 @@ is_vnode_prov_done(char * vnode)
 	set_task(WORK_Immed, 0, do_provisioning, (void *) NULL);
 }
 
+/**
+ * @brief
+ *	Call check_and_run_jobs(prov_vnode_info) from a timed task to ensure that
+ *	hooks are synced with the node before job get start
+ *
+ * @param[in,out]       ptask - work task structure contains prov_vnode_info
+ */
+
+static void 
+is_sync_hooks_done(struct work_task *ptask)
+{
+	struct  pbsnode         *pnode = NULL;
+	struct prov_vnode_info * prov_vnode_info;
+
+	prov_vnode_info = (struct prov_vnode_info *) ptask->wt_parm1;
+	pnode = (struct pbsnode *) find_nodebyname(prov_vnode_info->pvnfo_vnode);
+
+	DBPRT(("%s: calling [check_and_run_jobs] from is_sync_hooks_done\n", __func__))
+
+	check_and_run_jobs(prov_vnode_info);
```

Comment:
What will happen if the syncing is still not finished and check_and_run_jobs returns -5 again here?